File: src/main/java/com/camackenzie/exvi/core/model/ExerciseType.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.camackenzie.exvi.core.model;

/**
 *
 * @author callum
 */
public enum ExerciseType {
    STRENGTH,
    WARMUP,
    COOLDOWN,
    PLYOMETRIC,
    CONDITIONING,
    POWER_LIFTING;

    public static ExerciseType fromString(String s) {
        return EnumUtils.enumFromString(ExerciseType.class, s);
    }

    @Override
    public String toString() {
        return EnumUtils.formatName(super.toString());
    }
}

File: src/commonMain/kotlin/com/camackenzie/exvi/core/model/ExerciseType.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.camackenzie.exvi.core.model

import com.camackenzie.exvi.core.model.EnumUtils.formatName
import com.camackenzie.exvi.core.model.EnumUtils

/**
 *
 * @author callum
 */
@kotlinx.serialization.Serializable
enum class ExerciseType {
    STRENGTH, WARMUP, COOLDOWN, PLYOMETRIC, CONDITIONING, POWER_LIFTING;

    override fun toString(): String {
        return formatName(super.toString())
    }

    companion object {
        fun fromString(s: String): ExerciseType? {
            return EnumUtils.enumFromString<ExerciseType>(s)
        }
    }
}